pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -874,6 +871,8 @@ public void ensureAllPartitionsConsuming(TableConfig tableConfig, PartitionLevel
     HelixHelper.updateIdealState(_helixManager, realtimeTableName, idealState -> {
       assert idealState != null;
       if (idealState.isEnabled()) {
+        new MissingConsumingSegmentFinder(realtimeTableName, _propertyStore, _controllerMetrics)
```

Comment:
should this be run after the validation manager is done with the current round? Bit confusing that we first try to emit for what's missing and then immediately proceed to fixing.

Reply:
The reason it was at the beginning of the validation manager was to avoid stream connection issues. One root cause for not having consuming segments in ideal state is that stream client can throw exception during stream metadata for creating new consuming segment either in 1) segment commit or in 2) validation manager for detecting new partition groups. By just looking at ideal state, we can detect for how long we haven't had a consuming segment for a partition.

Anyway I moved this SegmentStatusChecker job per Subbu's suggestion, so it runs more frequently.